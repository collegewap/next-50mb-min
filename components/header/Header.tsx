import React from 'react'
import Head from 'next/head'

export type HeaderProps = {
  title: string
  description: string
}

export const Header = ({ title, description }: HeaderProps): JSX.Element => {
  return (
    <div>
      <Head>
        <title>{title}</title>
        <meta name="description" content={description} />
      </Head>
    </div>
  )
}

export default Header
