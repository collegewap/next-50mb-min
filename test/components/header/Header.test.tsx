import React from 'react'
import { Header } from '../../../components/header/Header'
import { render } from '../../testUtils'

jest.mock('next/head', () => {
  return {
    __esModule: true,
    default: ({ children }: { children: Array<React.ReactElement> }) => {
      return <>{children}</>
    },
  }
})

describe('Header', () => {
  const headerProps = {
    title: 'This is a heading',
    description: 'This is a description',
  }
  it('matches snapshot', () => {
    const { asFragment } = render(<Header {...headerProps} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
  it('has a title tag', () => {
    const { container } = render(<Header {...headerProps} />, {})
    const title = container.getElementsByTagName('title')[0]

    expect(title).toBeTruthy()
    expect(title?.innerHTML).toBe(headerProps.title)
  })
  it('has a meta description tag', () => {
    const { container } = render(<Header {...headerProps} />, {})
    const description: HTMLMetaElement = container.querySelector(
      'meta[name=description]'
    )

    expect(description).toBeTruthy()
    expect(description.content).toBe(headerProps.description)
  })
})
