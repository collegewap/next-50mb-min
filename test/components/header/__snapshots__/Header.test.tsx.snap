// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <title>
      This is a heading
    </title>
    <meta
      content="This is a description"
      name="description"
    />
  </div>
</DocumentFragment>
`;
